app/controllers/idv/cancellations_controller.rb

```diff
@@ -10,21 +10,30 @@ def new
       analytics.idv_cancellation_visited(step: params[:step], **properties)
       self.session_go_back_path = go_back_path || idv_path
       @hybrid_session = hybrid_session?
+      @presenter = CancellationsPresenter.new(
+        sp_name: decorated_session.sp_name,
+        url_options: url_options,
+      )
     end
 
     def update
-      if params.key?(:cancel)
-        analytics.idv_cancellation_go_back(step: params[:step])
-        redirect_to session_go_back_path || idv_path
-      else
-        render :new
-      end
+      analytics.idv_cancellation_go_back(step: params[:step])
+      redirect_to session_go_back_path || idv_path
     end
 
     def destroy
       analytics.idv_cancellation_confirmed(step: params[:step])
-      @return_to_sp_path = return_to_sp_failure_to_proof_path(location_params)
-      @hybrid_session = hybrid_session?
+      cancel_session
+      if hybrid_session?
+        render :destroy
+      else
+        render json: { redirect_url: cancelled_redirect_path }
```

Comment:
Initially I wanted this to just be a plain redirect, but since this is submitted as a form, the redirect would be blocked due to CSP. Another option could be to have this be a `GET` request, but I think it makes sense the way it's implemented now, since it is technically a destructive action (cancelling = "destroying" the session).